Source/WebKit/Shared/Cocoa/CoreIPCNSURLRequest.mm

```diff
@@ -0,0 +1,321 @@
+    }
+
+    SET_NSURLREQUESTDATA_PRIMITIVE(startTimeoutTime, NSNumber, double);
+    SET_NSURLREQUESTDATA_PRIMITIVE(requiresShortConnectionTimeout, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(payloadTransmissionTimeout, NSNumber, double);
+
+    id allowedProtocolTypes = dict[@"allowedProtocolTypes"];
+    if (allowedProtocolTypes != nil && [allowedProtocolTypes isKindOfClass:[NSNumber class]]) {
+        auto val = [allowedProtocolTypes unsignedCharValue];
+        if (isValidEnum<NSURLRequestAllowedProtocolTypes>(val))
+            m_data.allowedProtocolTypes = static_cast<NSURLRequestAllowedProtocolTypes>(val);
+    }
+
+    SET_NSURLREQUESTDATA(boundInterfaceIdentifier, NSString, CoreIPCString);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsExpensiveNetworkAccess, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsConstrainedNetworkAccess, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsUCA, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(assumesHTTP3Capable, NSNumber, bool);
+
+    id attribution = dict[@"attribution"];
+    if (attribution != nil && [attribution isKindOfClass:[NSNumber class]]) {
+        auto val = [allowedProtocolTypes unsignedCharValue];
+        if (isValidEnum<NSURLRequestAttribution>(val))
+            m_data.attribution = static_cast<NSURLRequestAttribution>(val);
+    }
+
+    SET_NSURLREQUESTDATA_PRIMITIVE(knownTracker, NSNumber, bool);
+    SET_NSURLREQUESTDATA(trackerContext, NSString, CoreIPCString);
+    SET_NSURLREQUESTDATA_PRIMITIVE(privacyProxyFailClosed, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(privacyProxyFailClosedForUnreachableNonMainHosts, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(privacyProxyFailClosedForUnreachableHosts, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(requiresDNSSECValidation, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowsPersistentDNS, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(prohibitPrivacyProxy, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(allowPrivateAccessTokensForThirdParty, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(useEnhancedPrivacyMode, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(blockTrackers, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(failInsecureLoadWithHTTPSDNSRecord, NSNumber, bool);
+    SET_NSURLREQUESTDATA_PRIMITIVE(isWebSearchContent, NSNumber, bool);
+
+    NSArray *contentDispositionEncodingFallbackArray = dict[@"contentDispositionEncodingFallbackArray"];
+    if (contentDispositionEncodingFallbackArray != nil && [contentDispositionEncodingFallbackArray isKindOfClass:[NSArray class]]) {
+        Vector<CoreIPCNumber> vector;
+        vector.reserveInitialCapacity(contentDispositionEncodingFallbackArray.count);
+        for (id element in contentDispositionEncodingFallbackArray) {
+            if ([element isKindOfClass:[NSNumber class]]) {
+                CoreIPCNumber tooAdd(element);
+                vector.append(WTFMove(tooAdd));
```

Comment:
I think there's an extra o in too
This could be a one liner:
vector.append(CoreIPCNumber(element))